--- lib/csp/meson.build	2024-03-05 19:26:11.487329692 +0100
+++ lib/csp/meson.build	2024-03-05 19:26:01.151385413 +0100
@@ -71,10 +71,23 @@
   install_headers(csp_config_h, install_dir : 'include/csp')
 endif
 
+freertos_src = ['freertos/list.c',
+                'freertos/queue.c',
+                'freertos/tasks.c',
+                'freertos/timers.c',
+                'freertos/portable/GCC/ARM_CM4F/port.c',
+               ]
+
+freertos_inc = include_directories(
+    'freertos/include',
+	'freertos/portable/GCC/ARM_CM4F',
+)
+
+csp_sources += freertos_src
 
 csp_lib = library('csp',
 	sources: [csp_sources, csp_config_h],
-	include_directories : include_directories('include', 'src'),
+	include_directories : [include_directories('include', 'src'), freertos_inc],
 	dependencies : csp_deps,
 	c_args : csp_c_args,
 	install : false,
@@ -84,7 +97,7 @@
 # The following dependency variable is for parent projects to link
 # against libcsp. https://mesonbuild.com/Subprojects.html
 csp_dep = declare_dependency(
-	include_directories : csp_inc,
+	include_directories : [csp_inc, freertos_inc],
 	link_with : csp_lib,
 	dependencies : csp_deps,
 )
